### PR TITLE
modified loop and conditional code

### PR DESCRIPTION
**Pull request by ayodeji-ti**

### CODE REVIEW

File: src/utils/code.ts [hunk 1]
```diff
@@ -103,10 +103,10 @@ const buildStyledProps = (propsNames: string[], childComponent: IComponent) => {
   return propsContent
 }
 
-const returnConditionalValue = (
+const returnConditionalValue: (
   propsNames: string[],
   childComponent: IComponent,
-) => {
+) => any = (propsNames: string[], childComponent: IComponent) => {
```
Comment (vinayak-sachdeva) [inline src/utils/code.ts]: why any?

File: src/utils/code.ts [hunk 2]
```diff
@@ -184,10 +184,20 @@ const buildSingleBlock = ({
       ${buildBlock({ component: childComponent, components, forceBuildBlock })}
       </${componentName}>`
     } else if (componentName === 'Conditional') {
-      content += `{${returnConditionalValue(
-        propsNames,
-        childComponent,
-      )}? <>${buildSingleBlock({
+      content += `{${
+        returnConditionalValue(propsNames, childComponent) === true ||
```
Comment (vinayak-sachdeva) [inline src/utils/code.ts]: why not typeof?

File: src/utils/code.ts [hunk 3]
```diff
@@ -184,10 +184,20 @@ const buildSingleBlock = ({
...
+        returnConditionalValue(propsNames, childComponent) === false
```
Comment (vinayak-sachdeva) [inline src/utils/code.ts]: Extract the value in variable first. See if you can improve the code, its not readable

File: src/utils/code.ts [hunk 4]
```diff
@@ -199,10 +209,15 @@ const buildSingleBlock = ({
         forceBuildBlock,
       })}</>}`
     } else if (componentName === 'Loop') {
-      content += `{Object.values({list: ${returnLoopValue(
-        propsNames,
-        childComponent,
-      )}})[0].map((item${childComponent.id.slice(
+      content += `{${
+        typeof returnLoopValue(propsNames, childComponent) === 'object'
```
Comment (vinayak-sachdeva) [inline src/utils/code.ts]: same comments as above

File: src/utils/code.ts [hunk 5]
```diff
@@ -273,10 +288,22 @@ const buildBlock = ({
       ${buildBlock({ component: childComponent, components, forceBuildBlock })}
       </${componentName}>`
       } else if (componentName === 'Conditional') {
-        content += `{${returnConditionalValue(
-          propsNames,
-          childComponent,
-        )}? <>${buildSingleBlock({
+        content += `{${
+          returnConditionalValue(propsNames, childComponent) === true ||
```
Comment (vinayak-sachdeva) [inline src/utils/code.ts]: same as above

File: src/utils/code.ts [hunk 6]
```diff
@@ -288,10 +315,18 @@ const buildBlock = ({
           forceBuildBlock,
         })}</>}`
       } else if (componentName === 'Loop') {
-        content += `{[${returnLoopValue(
-          propsNames,
-          childComponent,
-        )}].map((item${childComponent.id.slice(
+        content += `{${
+          typeof returnLoopValue(propsNames, childComponent) === 'object'
```
Comment (vinayak-sachdeva) [inline src/utils/code.ts]: same as above. Try to refactor duplicates

**Review by vinayak-sachdeva — CHANGES_REQUESTED** [review]

File: src/utils/code.ts (hunk 1 above)
Comment (ayodeji-ti) [reply src/utils/code.ts]: its possible the type is a string, I'll change it to boolean | string

**Review by vinayak-sachdeva — APPROVED** [review]

**Pull request merged**